### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by Sem-Chehovich**

### CODE REVIEW

File: src/app/core/components/breadcrumbs/breadcrumbs.component.html [hunk 1]
```diff
@@ -1 +1,3 @@
-<p>breadcrumbs works!</p>
+<div class="breadcrumb-container">
+    <p>Courses</p>
```
Comment (FedorovAlexander) [inline src/app/core/components/breadcrumbs/breadcrumbs.component.html]: It would be better to use `ul` tag for breadcrumbs list

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
 import { AppComponent } from './app.component';
+import { CoreModule } from './core/core.module';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatIconModule } from '@angular/material/icon';
+// import { DurationPipe } from './shared/pipes/duration.pipe';
 
 @NgModule({
   declarations: [
     AppComponent,
+    // DurationPipe,
```
Comment (FedorovAlexander) [inline src/app/app.module.ts]: The best practice is to create Shared Module (in your shared folder) and put all pipes, directives, and util functions there. Then, instead of importing elements one by one, you import only one Shared module.

File: src/app/core/components/header/header.component.html [hunk 3]
```diff
@@ -1 +1,10 @@
-<p>header works!</p>
+<header class="header-container">
+    <div class="logo-container">
+        <img src="../../../../assets/logo.svg" alt="logo">
+        <div class="logo-item">VIDEO COURSE</div>
+    </div>
+    <div class="header-actions">
+        <a><img src="../../../../assets/ic-user.svg" alt="user logo"> User Info</a>
```
Comment (FedorovAlexander) [inline src/app/core/components/header/header.component.html]: `<button>` is more suitable in this case

File: src/app/courses/courses.module.ts [hunk 4]
```diff
@@ -16,10 +22,13 @@ import { SearchCourseComponent } from './components/search-course/search-course.
     CourseCardComponent,
     ModalWindowComponent,
     SearchCourseComponent,
+    DurationPipe,
```
Comment (FedorovAlexander) [inline src/app/courses/courses.module.ts]: Pipes, directives, or other util functions should be imported with the Shared module.

File: src/app/courses/pages/courses-page/courses-page.component.html [hunk 5]
```diff
@@ -1 +1,13 @@
...
+  <app-search-course></app-search-course>
+  <div class="add-course">
+    <button class="btn">Add Course</button>
+  </div>
+  <ng-container *ngFor="let course of courses">
+    <app-course-card [course]="course" (editCourse)="onEditCourse($event)" (deleteCourse)="onDeleteCourse($event)">
+    </app-course-card>
+  </ng-container>
+  <div (click)="loadMore()" class="courses-load-more">
+    <a routerLink="/">LOAD MORE</a>
```
Comment (FedorovAlexander) [inline src/app/courses/pages/courses-page/courses-page.component.html]: This button loads additional courses. There shouldn't be `routerLink`. `<button>` is more suitable here.

File: src/app/courses/pages/courses-page/courses-page.component.html [hunk 6]
```diff
@@ -1 +1,13 @@
-<p>courses-page works!</p>
+<div class="courses-container">
+  <app-search-course></app-search-course>
+  <div class="add-course">
+    <button class="btn">Add Course</button>
```
Comment (FedorovAlexander) [inline src/app/courses/pages/courses-page/courses-page.component.html]: An icon is missed

File: src/app/courses/pages/courses-page/courses-page.component.html [hunk 7]
```diff
@@ -1 +1,13 @@
...
+  </div>
+  <ng-container *ngFor="let course of courses">
```
Comment (FedorovAlexander) [inline src/app/courses/pages/courses-page/courses-page.component.html]: It would be better to create a separate component CoursesList.
For example, if in the future you will need a courses list on another page, now you have to copy and paste all related to courses list logic from the courses page to another page.

If you have a separate CoursesList component, you just need to reuse the CoursesList component on another page.

Also, using `trackBy` function is mandatory for this task.

File: src/app/courses/pages/courses-page/courses-page.component.ts [hunk 8]
```diff
@@ -1,15 +1,51 @@
...
   }
 
+  onEditCourse(courseId: number) {
+    console.log(`Edit Course with Id: ${courseId}`);
+  }
+
+  onDeleteCourse(courseId: number) {
+    console.log(`Delete Course with Id: ${courseId}`);
+  }
+
```
Comment (FedorovAlexander) [inline src/app/courses/pages/courses-page/courses-page.component.ts:16]: Could you please put mocks for courses and edit/delete methods to the separate (CoursesList) component?

**Review by FedorovAlexander — CHANGES_REQUESTED** [review]
Good Job! Still, there are some issues to fix.

File: src/app/courses/components/course-card/course-card.component.html [hunk 9]
```diff
@@ -1 +1,22 @@
-<p>course-card works!</p>
+<div class="card-container">
```
Comment (FedorovAlexander) [inline src/app/courses/components/course-card/course-card.component.html]: You can use `<mat-card>` here. It will add a box-shadow, that is currently not implemented.

File: src/app/courses/components/course-card/course-card.component.scss [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    .btn-container {
+      display: flex;
+      height: 30px;
+      align-self: center;
+      justify-content: flex-end;
+
+      .btn {
+        display: flex;
```
Comment (FedorovAlexander) [inline src/app/courses/components/course-card/course-card.component.scss:58]: Could you please make left and right padding of the buttons visually equal?

**Review by FedorovAlexander — CHANGES_REQUESTED** [review]

File: src/app/core/components/breadcrumbs/breadcrumbs.component.html (hunk 1 above)
Comment (Sem-Chehovich) [reply src/app/core/components/breadcrumbs/breadcrumbs.component.html]: fixed thanks

File: src/app/app.module.ts (hunk 2 above)
Comment (Sem-Chehovich) [reply src/app/app.module.ts]: fixed thanks

File: src/app/core/components/header/header.component.html (hunk 3 above)
Comment (Sem-Chehovich) [reply src/app/core/components/header/header.component.html]: fixed thanks

File: src/app/courses/courses.module.ts (hunk 4 above)
Comment (Sem-Chehovich) [reply src/app/courses/courses.module.ts]: fixed thanks

File: src/app/courses/pages/courses-page/courses-page.component.html (hunk 5 above)
Comment (Sem-Chehovich) [reply src/app/courses/pages/courses-page/courses-page.component.html]: fixed thanks

File: src/app/courses/pages/courses-page/courses-page.component.html (hunk 6 above)
Comment (Sem-Chehovich) [reply src/app/courses/pages/courses-page/courses-page.component.html]: fixed thanks

File: src/app/courses/pages/courses-page/courses-page.component.html (hunk 7 above)
Comment (Sem-Chehovich) [reply src/app/courses/pages/courses-page/courses-page.component.html]: fixed thanks

File: src/app/courses/pages/courses-page/courses-page.component.ts (hunk 8 above)
Comment (Sem-Chehovich) [reply src/app/courses/pages/courses-page/courses-page.component.ts:16]: fixed thanks

File: src/app/courses/components/course-card/course-card.component.html (hunk 9 above)
Comment (Sem-Chehovich) [reply src/app/courses/components/course-card/course-card.component.html]: fixed thanks

File: src/app/courses/components/course-card/course-card.component.scss (hunk 10 above)
Comment (Sem-Chehovich) [reply src/app/courses/components/course-card/course-card.component.scss:58]: fixed thanks

File: src/app/courses/components/course-card/course-card.component.html (hunk 9 above)
Comment (FedorovAlexander) [reply src/app/courses/components/course-card/course-card.component.html:1]: Sorry, I was not clear here.
There should be a good reason for adding a library to your application. If you add angular material, it would be better to use its components. Adding such a big library as angular material only for using icons is not reasonable.

Course card is a perfect case for `mat-card`, so I suggest using it. Same with a modal dialog (you will create it later in the course).

To summarize: it's not about box-shadow. It's about using the library.

File: src/app/courses/components/course-card/course-card.component.scss [hunk 11]
```diff
@@ -5,6 +5,7 @@
   border-radius: 8px;
   background-color: #ffffff;
   margin-bottom: 24px;
+  box-shadow: 0 1px 2px 0 rgba(29, 30, 38, 0.10), 0 2px 10px 0 rgba(29, 30, 38, 0.15);
```
Comment (FedorovAlexander) [inline src/app/courses/components/course-card/course-card.component.scss:10]: The best practice is to use angular material `mat-card` component in this case. This box-shadow is not necessary.

File: src/app/courses/components/courses-list/courses-list.component.html [hunk 12]
```diff
@@ -0,0 +1,4 @@
+<ng-container *ngFor="let course of courses">
+    <app-course-card [course]="course" (editCourse)="onEditCourse($event)" (deleteCourse)="onDeleteCourse($event)">
```
Comment (FedorovAlexander) [inline src/app/courses/components/courses-list/courses-list.component.html]: Don't forget about semantics. As far as it is a list, it would be better to use `ul` and `li` here.

Also, using trackBy function is mandatory.
https://codebriefly.com/how-to-use-trackby-in-angular-with-example/

**Review by FedorovAlexander — CHANGES_REQUESTED** [review]

File: src/app/courses/pages/courses-page/courses-page.component.html (hunk 7 above)
Comment (FedorovAlexander) [reply src/app/courses/pages/courses-page/courses-page.component.html]: You forgot about trackBy.

File: src/app/courses/components/course-card/course-card.component.html (hunk 9 above)
Comment (FedorovAlexander) [reply src/app/courses/components/course-card/course-card.component.html]: Sorry, I was not clear here.

There should be a good reason to add a library to your application. If you add angular material, it would be better to use its components. Adding such a big library only for icons is not reasonable.

Course card is the perfect use case for `mat-card`, so I suggest using it. Same with modal-dialog component (you will create it later in the course). Of course, if you do not want to use angular material you can remove from package.json it and rewrite the code related to icons. It's up to you.

To summarize: it's not about box shadow. It's about using the library.

File: src/app/courses/components/course-card/course-card.component.scss (hunk 11 above)
Comment (Sem-Chehovich) [reply src/app/courses/components/course-card/course-card.component.scss:10]: fixed

File: src/app/courses/components/course-card/course-card.component.html (hunk 9 above)
Comment (Sem-Chehovich) [reply src/app/courses/components/course-card/course-card.component.html]: fixed

File: src/app/courses/components/courses-list/courses-list.component.html (hunk 12 above)
Comment (Sem-Chehovich) [reply src/app/courses/components/courses-list/courses-list.component.html]: fixed

Comment (FedorovAlexander) [reply src/app/courses/components/courses-list/courses-list.component.html]: trackBy is not implemented

**Review by FedorovAlexander — CHANGES_REQUESTED** [review]

File: src/app/core/components/breadcrumbs/breadcrumbs.component.scss [hunk 13]
```diff
@@ -0,0 +1,11 @@
+.breadcrumb-container {
+    width: 100%;
+    height: 64px;
+    background-color: #fff;
+
+    li {
+        padding-top: 20px;
+        padding-left: 85px;
+        list-style-type: none;
```
Comment (FedorovAlexander) [inline src/app/core/components/breadcrumbs/breadcrumbs.component.scss]: Could you please remove lists padding-left and make the breadcrumbs item text bold as it is shown in the mockup?

**Review by FedorovAlexander — CHANGES_REQUESTED** [review]

**Review by FedorovAlexander — APPROVED** [review]
